MediaGalleryRenditions/Plugin/SetPathOnInsert.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryRenditions\Plugin;
+
+use Magento\Cms\Controller\Adminhtml\Wysiwyg\Images\OnInsert;
+use Magento\Cms\Helper\Wysiwyg\Images;
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Controller\Result\RawFactory;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Framework\Filesystem;
+use Magento\Framework\Filesystem\Directory\ReadInterface;
+use Magento\MediaGalleryRenditionsApi\Api\GetRenditionPathInterface;
+use Magento\MediaContentApi\Model\Config;
+
+/**
+ * Set renditions path on content insert
+ */
+class SetPathOnInsert
+{
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var GetRenditionPathInterface
+     */
+    private $getRenditionPath;
+
+    /**
+     * @var RawFactory
+     */
+    private $resultRawFactory;
+
+    /**
+     * @var Images
+     */
+    private $imagesHelper;
+
+    /**
+     * @var Config
+     */
+    private $config;
+
+    /**
+     * SetPathOnInsert constructor.
+     * @param GetRenditionPathInterface $getRenditionPath
+     * @param RawFactory $resultRawFactory
+     * @param Filesystem $filesystem
+     * @param Images $imagesHelper
+     * @param Config $config
+     */
+    public function __construct(
+        GetRenditionPathInterface $getRenditionPath,
+        RawFactory $resultRawFactory,
+        Filesystem $filesystem,
+        Images $imagesHelper,
+        Config $config
+    ) {
+        $this->getRenditionPath = $getRenditionPath;
+        $this->resultRawFactory = $resultRawFactory;
+        $this->filesystem = $filesystem;
+        $this->imagesHelper = $imagesHelper;
+        $this->config = $config;
+    }
+
+    /**
+     * Set Rendition's path on content insert
+     *
+     * @param OnInsert $subject
+     * @param callable $proceed
+     * @return ResultInterface
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function aroundExecute(OnInsert $subject, callable $proceed): ResultInterface
```

Comment:
I think it might be better to extract the logic from `\Magento\Cms\Controller\Adminhtml\Wysiwyg\Images\OnInsert::execute` to a model (creating a related `magento2` PR) and add a before plugin for that model changing the `filename`.